### ISSUE TITLE
What does "support the following text style properties" mean (1.4.12)?

### ISSUE THREAD

**Issue by JAWS-test** [w3c/wcag#884]
If I have a closed environment where the developer can set all the text style properties, but the user can't (e.g. browser in a company where the use of user styles is not possible), what then?
* does the developer have to implement a setting possibility for the user, so that she or he can adjust the text style properties? or
* is 1.4.12 not applicable because the user cannot adjust the text style properties (without the help of the developer)?

**Comment by patrickhlauke** [comment]
the latter

**Comment by awkawk** [comment]
@patrickhlauke I'd say the former. If a user has a locked-down version of a browser that is required for use in company X then the author needs to provide this functionality. It is just like resize text where the Understanding document says:

> The author cannot rely on the user agent to satisfy this Success Criterion for HTML content if users do not have access to a user agent with zoom support. For example, if they work in an environment that requires them to use IE 6.
> 
> If the author is using a technology whose user agents do not provide zoom support, the author is responsible to provide this type of functionality directly or to provide content that works with the type of functionality provided by the user agent. If the user agent doesn't provide zoom functionality but does let the the user change the text size, the author is responsible for ensuring that the content remains usable when the text is resized.

**Comment by patrickhlauke** [comment]
> If the author is using a technology whose user agents do not provide zoom support

the technology in this case (i'm assuming) is HTML/CSS, and default user agents provide support for this. the IE6 case (for text size) was specifically of its time, and related to not being able to rely on IE6 resizing text set in pixel sizes...I think that clause is very much "of its time" and mainly due to the preponderance of IE use at the time.

if the environment that the user has to use uses some cut-down browser (like a point of sales type device for instance), is this a web content author responsibility, or a user agent responsibility?

**Comment by patrickhlauke** [comment]
or are you suggesting that it's the web author's responsibility now that for a point-of-sales type environment, THEY must provide / backfill anything that was not provided by default in the user agent or THEY fail WCAG? seems a stretch to me...

**Comment by awkawk** [comment]
> if the environment that the user has to use uses some cut-down browser (like a point of sales type device for instance), is this a web content author responsibility, or a user agent responsibility?

I'd say author. Otherwise the author can't make a conformance claim that has them relying on accessibility-supported technology.

**Comment by patrickhlauke** [comment]
to split hairs, for 1.4.12

> In content implemented using markup languages that support the following text style properties, no loss of content or functionality occurs by setting all of the following and by changing no other style property

it only normatively mandates that the content *supports* the text properties, not that a user must be able to actually set them, so not quite sold on that.

for 1.4.4

> Except for captions and images of text, text can be resized without assistive technology up to 200 percent without loss of content or functionality.

ok, *can be resized*. though assuming this closed environment is a full system (including software and hardware etc), i'd still tend to say it's not a failure of WCAG, but of UAAG https://www.w3.org/TR/UAAG20/#gl-text-config (end result would still be that the system as a whole fails this aspect). it becomes even blurrier if the author doesn't know in advance if their content is or isn't going to be used in a closed environment...

**Comment by awkawk** [comment]
> it only normatively mandates that the content supports the text properties, not that a user must be able to actually set them, so not quite sold on that.

It normatively requires that making those changes doesn't result in loss of content or functionality. Even viewed this way, your point about not requiring the user to be able to set is taken.

For 1.4.4 the second paragraph quoted above from the understanding document indicates that the WG viewed this as an author responsibility if the user agent didn't handle it.

**Comment by patrickhlauke** [comment]
> It normatively requires that making those changes doesn't result in loss of content or functionality

that's what i meant, but my wording was probably unclear. to be clearer, i meant "it doesn't mandate that the author must provide a mechanism to set them".

> if the user agent didn't handle it

fair enough, i guess i'm spoilt from not working in closed environments with custom user agents that offer no resizing for HTML/CSS content (which again i'll assume are things like a POS)

**Comment by JAWS-test** [comment]
If what I learned from @patrickhlauke is correct, it is what is in the normative SC that matters, not what is in the non-normative Understanding. Thus, there is an important difference between 1.4.4 and 1.4.12. 

* 1.4.4 knows no exception regarding the support of the font size adjustment / zoom function. 
* 1.4.12 explicitly names exceptions in two places:

> In content **implemented using markup languages that support** the following text style properties

> Exception: Human languages and scripts that do not make use of one or more of these text style properties in written text can conform using only the properties that exist for that combination of language and script.

Therefore I think that in 1.4.4 it is the responsibility of the author in a closed environment to implement the zoom functionality herself/himself. This is also the view of EN 301 549, which transposes WCAG 2.1. into European law.

For 1.4.12 the EN 301 549 is unfortunately not so clear and therefore my question.

**Comment by jake-abma** [comment]
I'm with the latter also:

The normative text doesn't suggest anything about author responsibility to provide a mechanism, also in the understanding there's:

> Author Responsibility
> The author requirement is both to not interfere with a user's ability to override the author settings, and to ensure that content thus modified does not break content in the manners shown in figures 1 through 3 in Effects of Not Allowing for Spacing Override.

"blocking" or providing the "wrong / not usable" UA is not the problem for the author to solve.

**Comment by JAWS-test** [comment]
I would like to ask you to add the outcome of this discussion to the Understanding document

**Comment by alastc** [comment]
HI @JAWS-test, the snippet Jake quoted was from the understanding doc for 1.4.12, what else do you think would make it clearer?

**Comment by JAWS-test** [comment]
I thought about it again and would leave the understanding as it is. I think this makes sense because in a closed environment the developer of the web software does not violate 1.4.12, but the employer, for example, who has closed the environment in such a way that a user cannot make the necessary adjustments.
